compiler/rustc_const_eval/src/transform/promote_consts.rs

```diff
@@ -457,7 +457,9 @@ impl<'tcx> Validator<'_, 'tcx> {
         match kind {
             // Reject these borrow types just to be safe.
             // FIXME(RalfJung): could we allow them? Should we? No point in it until we have a usecase.
-            BorrowKind::Shallow | BorrowKind::Unique => return Err(Unpromotable),
+            BorrowKind::Shallow | BorrowKind::Mut { kind: MutBorrowKind::ClosureCapture } => {
+                return Err(Unpromotable);
+            }
```

Comment:
please exhaustively match on the remaining two kinds in `BorrowKind::Mut` and add a FIXME for a future PR:

> consider changing this to only promote `&mut []` for default borrows, also forbidding two phase borrows